src/wx/widgets/sdljoy.cpp

```diff
@@ -312,19 +309,37 @@ void wxSDLJoy::ConnectController(uint8_t joy)
 {
     SDL_Joystick* js_dev = nullptr;
     bool is_gc;
+    std::string name;
 
     if ((is_gc = SDL_IsGameController(joy))) {
         auto dev = SDL_GameControllerOpen(joy);
 
         if (dev) {
-            joystate[joy].dev = dev;
-
+            name   = SDL_GameControllerName(dev);
             js_dev = SDL_GameControllerGetJoystick(dev);
+
+            std::string extra_msg;
+
+            if (contains(JOYSTICK_OVERRIDES, name)) {
+                is_gc             = false;
```

Comment:
What does this logic mean?  Open as a gamecontroller, and then treat it as a joystick.

Reply:
Why is it necessary for these types of controllers?

Reply:
As I said in the commit message, they only send joystick events.

Reply:
Could you just open it as a joystick then?  For example:

If !contains(JOYSTICK_OVERRIDES, name) && IsGameController

else, open as a joystick

seems like the logic will be simpler

Reply:
I think I answered my own question, you're getting the name after it's open.  Maybe it's not possible to do what I'm saying.

Reply:
For some reason it put this in another comment.  You would need to instead use this function, so you don't have to open the controller just to find the name:

SDL_GameControllerNameForIndex(i)

But, I'll leave that decision up to you.

Reply:
For some reason I did not get emails for your comments, I will make this change.